pkg/asset/manifests/vsphere/infrastructure.go

```diff
@@ -77,8 +77,11 @@ func GetInfraPlatformSpec(ic *installconfig.InstallConfig, clusterID string) *co
 	platformSpec.APIServerInternalIPs = types.StringsToIPs(icPlatformSpec.APIVIPs)
 	platformSpec.IngressIPs = types.StringsToIPs(icPlatformSpec.IngressVIPs)
 	platformSpec.MachineNetworks = types.MachineNetworksToCIDRs(ic.Config.MachineNetwork)
-	platformSpec.MachineNetworks = append(platformSpec.MachineNetworks, vipsToCIDRs(ic.Config.VSphere.APIVIPs)...)
-	platformSpec.MachineNetworks = append(platformSpec.MachineNetworks, vipsToCIDRs(ic.Config.VSphere.IngressVIPs)...)
+
+	vips := getInstallConfigVIPs(ic)
+	for _, vip := range vips {
+		appendCIDRIfMissing(platformSpec.MachineNetworks, vipToCIDR(vip))
+	}
```

Comment:
```suggestion
	vips := getVIPsFromInstallConfig(ic)
	for _, vipCIDR := range vipsToCIDRs(vips) {
		if !slices.Contains(platformSpec.MachineNetworks, vipCIDR) {
			platformSpec.MachineNetworks = append(platformSpec.MachineNetworks, vipCIDR)
		}
	}
```

I think we can utilize `slices.Contains` here, right? This way, the change is more minimal.

Reply:
That said, we can keep the old func `vipsToCIDRs` that converts a slice of string to a slice of `configv1.CIDR`.